### DISCUSSION TITLE
NFS Server using Keepalived vIP

### DISCUSSION THREAD

**Discussion by KorbenGIT** [acassen/keepalived#2300] [Q&A]
Hello,

I'm trying to replicate this setup:

https://www.ianstacey.net/2021/07/20/basic-ha-nfs-server-with-keepalived/

So a simple active/passive NFS cluster via Keepalived.

But it doesn't work. The Keepalived configuration is valid and I got a vIP on the master node which can switch to the backup node.

But I cannot mount the NFS data folder using the Keepalived vIP:

mount -vvv -t nfs $vIP:/data ./nfs-test

returns mount.nfs: mount(2): "Connection timed out"

I've tested this setup on Debian 11, Debian 12 and OpenSuse LEAP 15. Same error on every OS.

It's the default NFS server configuration and Firewall rules on each OS. So the server is listening on everything 0.0.0.0 and all rules (INPUT/OUTPUT/FORWARD) are on ACCEPT.

Already tested with the NFSv4 on port 2049, got a different error, but NFS mountpoint still hangs indifinetly.

Does anybody have any idea how to make this setup works ?

**Comment by nser77** [comment]
I think you need to paste your ```keepalived``` configuration here if you want someone to help you.

**Reply by nser77** [reply]
And obviously specify if you are using NFS over TCP or UDP (pasting a ```tcpdump``` here could also be very helpful.).

**Comment by pqarmitage** [comment]
Closing due to no update for over 3 months.
